Ks/#250 replace short options

KSchopmeyer opened the pull request:
Changed a number of single character short option names to multicharacter shorter names. Changed the corresponding tests and added tests to be sure that the new short names are covered.

The list of options changes:

1. -p to  --pl for property list
2. --nq   --no-qualifiers for class
3. -R to  --rc result-class
4. -a to  --ac assoc-class
5. -C to  --rc result-class
6. -R to  --rr result-role
7. -q to  --iq include-qualifiers
8. -p to  --pl propertylis
9. -o to  --no for names-only
10. -c to --ico include-class-origin
11. -S to -s summary

The only issues I found in early testing is that it is easy to drop the first - and the one case where this causes problems is --no which becomes -n but fails because there is no namespace to not really a major issue.

andy-maier commented:
Comments:

* The following short option names should also become short long option names:
  - class enumerate CIM_Foo -n interop

Options:
  -d, --deep-inheritance        Include the complete subclass hierarchy of the
                                requested classes in the result set. Default:
                                Do not include subclasses.
  -l, --local-only

KSchopmeyer commented:
Updated with requested changes.  Made changes to docs/pywbemclicommands.rst.  i also made other changes to pywbemclicommands.rst behond just hte change from -d to -dk, -n to -no in that I 

1 instance create section and added more examples

2.  Changed the section that talked about CIM instance and instance names to cinclued classes since this affects classes.

3. Moved your paragraph on using backslash escap in shells to a footnote so it could be referenced multiple places. NOTE; The footnoe shouws up as section 4.9   since I have not figued out how to put in section header that you do not number.

andy-maier commented on docs/pywbemclicommands.rst:
```diff
@@ -535,42 +529,69 @@ The **instance** group defines commands that act on CIM instances including:
 
-  For a single property in the new instance this is simply the `--property`` option
-  with the property name and value:
+  The ``--property`` argument value component defines the value of the
```
What is a "argument value component"? Suggest to change to simply "option". It is clear from the context that it is actually the argument of the option that has the property value.

KSchopmeyer replied:
DONE

andy-maier commented on docs/pywbemclicommands.rst:
```diff
@@ -535,42 +529,69 @@ The **instance** group defines commands that act on CIM instances including:
-  with the property name and value:
+  The ``--property`` argument value component defines the value of the
+  property. Since the WBEM server (and pywbem) requires that each property be
+  typed, pywbemcli uses the CIMClass defined by CLASSNAME retrieved from the
```
On "pywbemcli uses the CIMClass defined by CLASSNAME retrieved from the WBEM server to define the type required to define the CIMProperty.":
Suggest to simplify to: 
"pywbemcli retrieves the CIM class to determine the property types."

KSchopmeyer replied:
DONE

andy-maier commented on docs/pywbemclicommands.rst:
```diff
@@ -535,42 +529,69 @@ The **instance** group defines commands that act on CIM instances including:
+  property. Since the WBEM server (and pywbem) requires that each property be
+  typed, pywbemcli uses the CIMClass defined by CLASSNAME retrieved from the
+  WBEM server to define the type required to define the CIMProperty. The value
+  argument contains just the value itself in the form of a string or numeric
```
What is "The value argument"?
Suggest to change to "The option argument".

KSchopmeyer replied:
DONE

andy-maier commented on docs/pywbemclicommands.rst:
```diff
@@ -535,42 +529,69 @@ The **instance** group defines commands that act on CIM instances including:
+  class (ex. -3 for Uint32).
+
+  See [#fbackslash]_ for information on use of backslashes in formating
+  property argument values:
```
Backslash-escaping the double quotes around string or datetime property values is not needed. The shell will make sure the (unescaped) double quoted string is passed as a single value to the option's variable.

andy-maier commented on docs/pywbemclicommands.rst:
```diff
@@ -535,42 +529,69 @@ The **instance** group defines commands that act on CIM instances including:
+    --property <property-name>=<property-value
+
+    where quotes around the value are only required if the value includes
+    whitespace.
```
I think instead of giving an (incomplete) description here, we should have a section "Specifying CIM property and parameter values" that is similar to "Specifying INSTANCENAME" and that defines the format precisely, e.g. by referring to DSP0004 stringValue etc.

KSchopmeyer replied:
Good idea. I was thinking about it also.  I added the section and referenced it from both the class and instance invokemethod definitions. NOTE: I put a TODO on this one since the first goal was to replace the short options and I need to do more cleanup on the text in this section.  We have this pr, the sections naming pr and your pywbmbemclicommands.rst pr that all get merged and then we can look at this chapter in more detail

KSchopmeyer replied:
DONE with 2 TODOs in the documentation to refactor the text.

andy-maier commented on docs/pywbemclicmdshelp.rst:
```diff
@@ -1948,11 +1950,13 @@ The following defines the help output for the `pywbemcli server get-centralinsts
+      --sp, --scoping-path CLASSLIST  Optional. Required only if profiles supports
                                       only scopig methodology. Multiples allowed
       -r, --reference-direction [snia|dmtf]
                                       Navigation direction for association.
```
docs/pywbemclicmdshelp.rst is out of date, which is th ereason it shows `-r` on `--reference-direction`.

KSchopmeyer replied:
DONE
